actors/common/src/main/java/org/sunbird/learner/util/Util.java

```
@@ -1463,24 +1467,86 @@ public static void checkProfileCompleteness(Map<String, Object> userMap) {
+        if (JsonKey.PRIVATE.equals(completeProfileVisibilityMap.get(field))) {
+          privateFieldsMap.put(field, userMap.get(field));
+        }
       }
       ElasticSearchUtil.upsertData(
           ProjectUtil.EsIndex.sunbird.getIndexName(),
           ProjectUtil.EsType.userprofilevisibility.getTypeName(),
           (String) userMap.get(JsonKey.USER_ID),
-          profileVisibilityMap);
-      UserUtility.updateProfileVisibilityFields(profileVisibilityMap, userMap);
+          privateFieldsMap);
+      UserUtility.updateProfileVisibilityFields(privateFieldsMap, userMap);
     } else {
       userMap.put(JsonKey.PROFILE_VISIBILITY, new HashMap<String, String>());
     }
   }
 
+  public static Map<String, String> getCompleteProfileVisibilityPrivateMap(
+      Map<String, String> userProfileVisibilityMap, ActorRef actorRef) {
+    Map<String, String> completeProfileVisibilityMap =
+        getCompleteProfileVisibilityMap(userProfileVisibilityMap, actorRef);
+    for (String key : completeProfileVisibilityMap.keySet()) {
+      if (JsonKey.PUBLIC.equals(completeProfileVisibilityMap.get(key))) {
+        completeProfileVisibilityMap.remove(key);
+      }
+    }
+    return completeProfileVisibilityMap;
+  }
+
+  public static Map<String, String> getCompleteProfileVisibilityMap(
+      Map<String, String> userProfileVisibilityMap, ActorRef actorRef) {
+    String defaultProfileVisibility =
+        ProjectUtil.getConfigValue(JsonKey.SUNBIRD_DEFAULT_USER_PROFILE_VISIBILITY);
+    if (!(defaultProfileVisibility.equals(JsonKey.PUBLIC)
+        || defaultProfileVisibility.equals(JsonKey.PRIVATE))) {
+      ProjectLogger.log(
+          "Util:getCompleteProfileVisibilityMap: Invalid configuration - "
+              + defaultProfileVisibility
+              + " - for default profile visibility(public/private)",
+          LoggerEnum.ERROR.name());
+      ProjectCommonException.throwServerErrorException(
```

We may not need to throw this error with a valid default.